### DISCUSSION TITLE
Problema DELETE su Prima Nota, errore 409

### DISCUSSION THREAD

**Discussion by Drakas** [fattureincloud/api#194] [Q&A] [answered]
Buonasera,

Non mi è possibile cancellare una voce da primanota. Dato l'id come da documentazione, ottengo il seguente errore:
**{"error":{"message":"Cannot perform the operation."}}**

E, dai log di Fatture in Cloud, ottengo: 

**DELETE | c/xxxxxx/cashbook/147xxxxx | 409**
Le x le ho inserite per anonimizzare. 

L'ID è corretto, ho anche controllato per sicurezza ed è anche identico a quando ispeziono la richiesta via browser del pannello di fatture in cloud: 
**https://secure.fattureincloud.it/edit_primanota.php?id=147xxxxx**

A cosa potrebbe essere dovuto?
Grazie e buona serata

**Comment by valmoz** [comment]
Buongiorno @Drakas 
La delete dei movimenti in prima nota è possibile solo se sono movimenti manuali. Se son movimenti derivati da fatture emesse/ricevute viene restituito 409

Grazie
Mauro

**Comment by Drakas** [comment, marked as answer]
Buongiorno @valmoz , 
i movimenti che cerco di cancellare sono proprio creati sempre via API, tramite endpoint cashbook con metodo POST.

Per questo che mi aspetterei di poterli cancellare senza ricevere quell'errore. Magari vi è qualche bug?

Grazie mille

**Reply by valmoz** [reply]
Buongiorno @Drakas 
no, non è un bug. Significa semplicemente che essendo legato a una fattura emessa o ricevuta, non è possibile cancellarlo.

Mauro

**Reply by Drakas** [reply]
Buona sera @valmoz,

Il punto è proprio che non c'è alcun legame di una fattura emessa o ricevuta nella voce che vorrei cancellare, allego prova.

La richiesta di creazione è:
```
{
  "data": {
    "date": "2022-12-16",
    "amount_out": 350,
    "payment_account_out": {
      "id": 17xxxx
    },
    "description": "xxx xxx xxxxxxxxx",
    "type": "out"
  }
}
```

Risposta alla corrispettiva creazione è:
```
{
  "data": {
    "id": 14xxxxx,
    "date": "2022-12-16",
    "amount_in": 0,
    "amount_out": 350,
    "description": "xxx xxx xxxxxxxxx",
    "issued_document_id": null,
    "received_document_id": null,
    "payment_account_in": null,
    "payment_account_out": {
      "id": 17xxxx,
      "name": "Cassa"
    },
    "kind": "cashbook"
  }
}
```


Tuttavia, se provo a cancellare questa voce, le API mi da errore 409.
Non dovrei poterla cancellare dato che non c'è alcun legame con le fatture?

**Reply by emajo** [reply]
Ciao @Drakas per eliminare il cashbook è necessario prependere la stringa CBI o CBO a seconda del tipo di cashbook all'id all'interno dell'url, per esempio in caso di un cashbook out l'url sarà` http://api-v2.local.fattureincloud.it/c/2/cashbook/CBO144947`.

Facci sapere se questo risolve il problema,
grazie,
Emanuele.

**Reply by Drakas** [reply]
Ciao @emajo, ti ringrazio proverò e verifico. Nel mentre, c'è scritta questa cosa nella documentazione ufficiale (https://developers.fattureincloud.it/api-reference#delete-/c/-company_id-/cashbook/-document_id-) ? Perché non ho letto di questo pre-append. Prossimi giorni provo :)

**Reply by emajo** [reply]
Dalle veloci prove che ho fatto sembrerebbe che quando viene creata una prima nota via api venga restituito l'id senza CBO/CBI, mentre facendo una list dovresti trovare gli id corretti, correggeremo il comportamento il prima possibile.

**Reply by Drakas** [reply]
Ciao @emajo , esattamente, proprio come da mia evidenza (https://github.com/fattureincloud/api/discussions/194#discussioncomment-4462490) dove riporto la risposta con l'ID di ritorno inizia con un numero e non lettere.

Taggo anche @valmoz per confermare che a questo punto è proprio un bug.

Purtroppo ad ora tutti gli ID stoccati di primanota nel sistema sono errati e non possono essere automazionati in futuro tramite referenza ID salvata. Mi taggate quando il bug è risolto?

Intanto grazie mille e buon anno

**Reply by Drakas** [reply]
Converto anche in issue per avere il workflow corretto

**Reply by emajo** [reply]
Perfetto, ti avviseremo appena risolviamo, per quanto riguarda quelli già salvati nel sistema si potrebbe andare ad agire con uno script che aggiunge CBI per i cashbook con type=in e CBO per i cashbook con type=out.

**Reply by emajo** [reply]
Fix rilasciato, non dovresti più riscontrare questo problema.
